File: worker/scripts/build-pages.py
#!/usr/bin/env python3
"""Build an Advanced Mode Cloudflare Pages deployment directory."""

from pathlib import Path
import shutil

ROOT = Path(__file__).resolve().parents[2]
PUBLIC = ROOT / "worker" / "public"
SOURCE = ROOT / "worker" / "src"
OUTPUT = ROOT / "dist"

if not PUBLIC.is_dir():
    raise SystemExit(f"Missing prepared public assets: {PUBLIC}")

shutil.rmtree(OUTPUT, ignore_errors=True)
shutil.copytree(PUBLIC, OUTPUT)
shutil.copy2(SOURCE / "worker.mjs", OUTPUT / "_worker.js")
shutil.copy2(SOURCE / "default-pages.mjs", OUTPUT / "default-pages.mjs")

print(f"Built Cloudflare Pages output: {OUTPUT}")

File: worker/scripts/build-pages.mjs
#!/usr/bin/env node
/** Build an Advanced Mode Cloudflare Pages deployment directory. */
import { cpSync, existsSync, rmSync } from 'node:fs';
import { dirname, join } from 'node:path';
import { fileURLToPath } from 'node:url';

const ROOT = join(dirname(fileURLToPath(import.meta.url)), '../..');
const PUBLIC = join(ROOT, 'worker/public');
const SOURCE = join(ROOT, 'worker/src');
const OUTPUT = join(ROOT, 'dist');

if (!existsSync(PUBLIC)) {
  console.error(`Missing prepared public assets: ${PUBLIC}`);
  process.exit(1);
}

rmSync(OUTPUT, { recursive: true, force: true });
cpSync(PUBLIC, OUTPUT, { recursive: true });
cpSync(join(SOURCE, 'worker.mjs'), join(OUTPUT, '_worker.js'));
cpSync(join(SOURCE, 'default-pages.mjs'), join(OUTPUT, 'default-pages.mjs'));
cpSync(join(SOURCE, 'invoice-logo-rgb.mjs'), join(OUTPUT, 'invoice-logo-rgb.mjs'));
cpSync(join(SOURCE, 'web-push-browser'), join(OUTPUT, 'web-push-browser'), { recursive: true });

console.log(`Built Cloudflare Pages output: ${OUTPUT}`);
